doc/source/whatsnew/v0.24.0.txt

```diff
@@ -322,6 +322,37 @@ is the case with :attr:`Period.end_time`, for example
 
    p.end_time
 
+.. _whatsnew_0240.api_breaking.sparse_values:
+
+``SparseArray`` is now an ``ExtensionArray``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+SparseArray is now implements the ExtensionArray interface.
+To conform to this interface, and for consistency with the rest of pandas, some API breaking
+changes were made:
+
+- ``SparseArray`` is no longer a subclass of :class:`numpy.ndarray`
+- ``SparseArray.dtype`` and ``SparseSeries.dtype`` are now instances of ``SparseDtype``, rather than ``np.dtype``. Access the underlying dtype with ``SparseDtype.subtype``.
+- :meth:`numpy.asarray(sparse_array)` now returns a dense array with all the values, not just the non-fill-value values (:issue:`14167`)
+- Passing a ``fill_value`` to ``SparseArray.take`` no longer implies ``allow_fill=True``.
+- ``SparseArray.take`` no longer accepts scalars for indices.
+- ``SparseArray.astype(sparse_dtype)`` will now change both the dtype of the underlying ``sp_values`` and the ``fill_value``. Previously, just
+  ``sparse_array.sp_values.dtype`` was changed. The same holds for a Series with spares values.
+- Setting ``SparseArray.fill_value`` to a fill value with a different dtype is now allowed.
+- The result of concatenating a mix of sparse and dense Series is a Series with sparse values.
+- ``SparseDataFrame.combine`` and ``DataFrame.combine_first`` no longer supports combining a spares column with a dense column while preserving
+  the sparse subtype. The result will be an object-dtype SparseArray.
+- Concatenating a SparseSeries and a dense series now returns a Series with sparse values.
+- ``SparseArray.take`` no longer accepts the ``out`` and ``mode`` parameters (previously, this raised if they were specified).
+
+
+Some new warnings are issued for operations that require or are likely to materialize a large dense array:
+
+- A :class:`errors.PerformanceWarning` is issued when using fillna with a ``method``, as a dense array is constructed to create the filled array. Filling with a ``value`` is the efficient way to fill a sparse array.
```

Comment:
It doesn't really fit the sparse model well. Consider

```
[1, x, x, x, 2, x, x, x]
```

If you ffill, there's not a good sparse value. You have 50% 1s and 50% 2s. With `value` setting `fill_value=value` maintains the same sparsity.

Reply:
OK, so that explanation is only relevant if the fill_value is a missing value (which is, eg, for integers not the case by default)

(anyhow, I am still wondering if `fillna` should try to be that 'smart' and change the fill_value)

Reply:
FWIW, I have no interest in *improving* SparseArray, beyond what happens as fallout from moving it to an ExtensionArray ;)